### PR TITLE
Convert fences to NDT_CONNECTED.

### PR DESCRIPTION
**Pull request by sofar**
This changes the drawtype of fences to NDT_CONNECTED
nodebox drawtype.

These nodes are drawn by the client with the needed
connections on the fly as the scene is drawn. There is
no logic needed by mods to modify the nodes.

These fences connect to (1) other fences, (2) planks
and (3) tree trunks, but nothing else. They do not
connect to stone, dirt, wool, etc. This is done by
the "connects_to" parameter, which takes groups and
node names.

This change requires minetest/minetest#3503.

### CODE REVIEW

**Comment by C1ffisme** [comment]
:+1: Very good. Let's kill the uncreative "fencelike" for good.

**Comment by sofar** [comment]
Removed spaces inside brackets.

**Comment by paramat** [comment]
Seems a good idea.

**Comment by sofar** [comment]
![]
(http://i.imgur.com/ViImmpj.png)

**Comment by paramat** [comment]
:+1:

File: mods/default/functions.lua [hunk 1]
```diff
@@ -213,16 +213,27 @@ function default.register_fence(name, def)
...
+			connect_n = {{-1/16,3/16,1/8,1/16,5/16,1/2},
+				{-1/16,-5/16,1/8,1/16,-3/16,1/2}},
+			connect_e = {{1/8,3/16,-1/16,1/2,5/16,1/16},
+				{1/8,-5/16,-1/16,1/2,-3/16,1/16}},
+			connect_w = {{-1/2,3/16,-1/16,-1/8,5/16,1/16},
+				{-1/2,-5/16,-1/16,-1/8,-3/16,1/16}},
+			connect_s = {{-1/16,3/16,-1/2,1/16,5/16,-1/8},
+				{-1/16,-5/16,-1/2,1/16,-3/16,-1/8}},
+		},
+		connects_to = {"group:fence", "group:wood", "group:tree"},
```
Comment (ShadowNinja) [inline mods/default/functions.lua]: It may make more sense for, eg, stone fences to connect to stone.  In fact, why not all solid nodes?

Comment (paramat) [inline mods/default/functions.lua]: I guess this is the default and stone fences can override with their own 'connects to'.

Comment (sofar) [inline mods/default/functions.lua]: walls are made out of stone:#796

https://github.com/sofar/minetest_game/commit/0c7d4418c9fa5fdacafa86490c7fd70b59ef45c4#diff-c967da67b272f4b656ff35c5d13941cdR33

Which is why I proposed to add these basic wall nodes :)

**Comment by Ekdohibs** [comment]
:+1:

**Comment by sofar** [comment]
- Updated with API changes

**Comment by paramat** [comment]
d979293f1448882e3a2eb0f7a3ecbaaf6b189d36

**Pull request closed without merging**